File: Algorithms/Easy/P1-Two_Sum/p1.ac.ts
function twoSum(nums: number[], target: number): number[] {
    let original: number[] = [...nums] // Clone nums
    nums.sort( (a, b) => a - b ) // Sort nums for searching
    let idx_down: number = 0, idx_up: number = nums.length - 1

    while(idx_down < idx_up) { // Assume exactly one solution
        let sum = nums[idx_down] + nums[idx_up]
        if (sum < target) ++idx_down
        else if (sum > target) --idx_up
        else break
    }

    let ans: number[] = []
    original.map( (value, index) => {
        if (value == nums[idx_down]) ans.push(index)
        else if (value == nums[idx_up]) ans.push(index)
    })

    return ans
};

File: Algorithms/Easy/P1-Two_Sum/p1.ac.go
func twoSum(nums []int, target int) []int {
    sorted := make([][2]int, len(nums))

    for i, num := range nums {
		sorted[i] = [2]int{i, num}
	}

    sort.Slice(sorted, func(i, j int) bool {
        return sorted[i][1] < sorted[j][1]
    })

    left, right := 0, len(nums) - 1

    for left < right {
        sum := sorted[left][1] + sorted[right][1]

        switch {
            case sum == target:
                return []int{sorted[left][0], sorted[right][0]}
            case sum < target:
                left++
                break
            case sum > target:
                right--
                break
        }
    }

    return []int{}
}